pkg/engine/internal/planner/physical/planner.go

```diff
@@ -370,35 +361,212 @@ func (p *Planner) processVectorAggregation(lp *logical.VectorAggregation, ctx *C
 		Operation: lp.Operation,
 	}
 	p.plan.graph.Add(node)
-	children, err := p.process(lp.Table, ctx)
+	child, err := p.process(lp.Table, ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err := p.plan.graph.AddEdge(dag.Edge[Node]{Parent: node, Child: child}); err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+func (p *Planner) hasNonMathExpressionChild(n Node) bool {
+	if n == nil {
+		return false
+	}
+
+	if _, ok := n.(*Projection); ok {
+		for _, c := range p.plan.Children(n) {
+			if p.hasNonMathExpressionChild(c) {
+				return true
+			}
+		}
+		return false
+	}
+
+	return true
+}
+
+func (p *Planner) processMathExpressionChild(c logical.Value, rootNode bool, ctx *Context) (Expression, Node, *ColumnExpr, error) {
```

Comment:
nit: `Context` is usually the first argument:

```suggestion
func (p *Planner) processMathExpressionChild(ctx *Context, c logical.Value, isRootNode bool) (Expression, Node, *ColumnExpr, error) {
```